src/Silex/ContentNegotiationProvider.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace eLife\ContentNegotiator\Silex;
+
+use eLife\ContentNegotiator\ContentNegotiator;
+use eLife\ContentNegotiator\Symfony\TypeAttributeValueResolver;
+use Negotiation\Accept;
+use Negotiation\AcceptCharset;
+use Negotiation\AcceptEncoding;
+use Negotiation\AcceptLanguage;
+use Negotiation\CharsetNegotiator;
+use Negotiation\EncodingNegotiator;
+use Negotiation\LanguageNegotiator;
+use Negotiation\Negotiator;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+
+final class ContentNegotiationProvider implements ServiceProviderInterface
+{
+    const ATTRIBUTE_ACCEPT = '_accept';
+    const ATTRIBUTE_ACCEPT_CHARSET = '_accept_charset';
+    const ATTRIBUTE_ACCEPT_ENCODING = '_accept_encoding';
+    const ATTRIBUTE_ACCEPT_LANGUAGE = '_accept_language';
+
+    public function register(Container $app)
+    {
+        $app['negotiator.accept'] = function () {
+            return new Negotiator();
+        };
+
+        $app['negotiator.accept_charset'] = function () {
+            return new CharsetNegotiator();
+        };
+
+        $app['negotiator.accept_encoding'] = function () {
+            return new EncodingNegotiator();
+        };
+
+        $app['negotiator.accept_language'] = function () {
+            return new LanguageNegotiator();
+        };
```

Comment:
not sure what the double configuration means `negotiator.*` vs `content_negotiator.*`, multiple versions support?

Reply:
`willdurand/negotiation` services vs wrappers provided by this library. Couldn't think of a better naming strategy, though aware that it's not that clear.

Reply:
Is it really necessary to expose the `willdurand/negotiation` in `$app`? Can they be hidden from clients instead? Perhaps prefix with `_` their names?

Reply:
No, could just remove them as services entirely.